chalice/app.py

```diff
@@ -379,6 +429,20 @@ def _register_view(view_func):
             return view_func
         return _register_view
 
+    def _validate_route_content_types(self, name, content_types):
+        # A routes content_types be homogeneous in their binary support.
+        binary, non_binary = [], []
+        for content_type in content_types:
+            if content_type in self.api.binary_types:
+                binary.append(content_type)
+            else:
+                non_binary.append(content_type)
+        if binary and non_binary:
+            raise ValueError(
+                'In view function "%s", the content_types %s support binary '
+                'and %s do not. All content_types must be consistent in their '
```

Comment:
Could you elaborate on how it change the interface? Seems like you should be able to say:

```
@app.route(..., content_types=['text/plain', 'image/png'])
def foo():
    if a:
        return Response('text content', headers={'Content-Type': 'text/plain')
    return Response(image_binary-data, headers={'Content-Type': 'img/png')
```